src/data/equiv/derangements/exponential.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Henry Swanson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henry Swanson, Patrick Massot
+-/
+import analysis.complex.basic
+import data.complex.exponential
+import data.equiv.derangements.finite
+import topology.metric_space.cau_seq_filter
+
+/-!
+# Derangement exponential series
+
+This file proves that the probability of a permutation on n elements being a derangement is 1/e.
+The specific lemma is `num_derangements_tendsto_e`.
+-/
+open filter
+open finset
+
+open_locale big_operators
+open_locale topological_space
+
+lemma complex.tendsto_iff_real (u : ℕ → ℝ) (x : ℝ) :
+  tendsto (λ n, u n) at_top (𝓝 x) ↔
+  tendsto (λ n, (u n : ℂ)) at_top (𝓝 (x : ℂ)) :=
+begin
+  split,
+  { exact λ h, (complex.continuous_of_real.tendsto x).comp h },
+  { exact λ h, (complex.continuous_re.tendsto x).comp h },
+end
+
+lemma complex.tendsto_exp_series (z : ℂ) :
+  tendsto (λ n, ∑ k in range n, z^k / k.factorial) at_top (𝓝 z.exp) :=
+begin
+  convert z.exp'.tendsto_limit,
+  unfold complex.exp,
+end
+
+lemma real.tendsto_exp_series (x : ℝ) :
+  tendsto (λ n, ∑ k in range n, x^k / k.factorial) at_top (𝓝 x.exp) :=
+begin
+  rw complex.tendsto_iff_real,
+  convert complex.tendsto_exp_series x; simp,
+end
```

Comment:
These should probably go elsewhere. With the definition of exp, maybe? The cau_seq and tendsto incompatibilities mean I didn't quite know what to do here for a while.